samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -213,32 +227,35 @@ public void onJobModelExpired() {
           if (SamzaContainerStatus.NOT_STARTED.equals(status) || SamzaContainerStatus.STARTED.equals(status)) {
             boolean shutdownComplete = false;
             try {
-              LOGGER.info("Shutting down container in onJobModelExpired for processor:" + processorId);
+              LOGGER.info("Shutting down the container: {} of stream processor: {} in onJobModelExpired.", container, processorId);
               container.pause();
               shutdownComplete = jcContainerShutdownLatch.await(taskShutdownMs, TimeUnit.MILLISECONDS);
-              LOGGER.info("ShutdownComplete=" + shutdownComplete);
+              LOGGER.info("Container shutdown completion status is: {} for stream processor: {}.", shutdownComplete, processorId);
+              if (scheduledSamzaContainerFuture != null && !scheduledSamzaContainerFuture.isDone()) {
```

Comment:
Where is this future field updated? I only see it set once in the beginning. What happens when a container is restarted?
Also, it's only volatile, so these operations aren't atomic. Does that matter here?

Reply:
This field is updated whenever SamzaContainer is started.
SamzaContainer is started only in the debounce thread (in response to handling the onNewJobModel zookeeper event).  
All runnable executions on the debounce thread are serialized(via DebounceScheduler task queue) and there're no expected concurrency problems.

volatile is used to address variable visibility problem.